autogen/agentchat/conversable_agent.py

```diff
@@ -1988,3 +2005,28 @@ def get_total_usage(self) -> Union[None, Dict[str, int]]:
             return None
         else:
             return self.client.total_usage_summary
+
+
+def register_function(
+    f: Callable[..., Any],
+    *,
+    agent: ConversableAgent,
```

Comment:
Rename it to "caller" to make it clear?

Reply:
Ok, but it still sounds wrong. The actual call is made by the user proxy (executor).